packages/extractor/src/api/extractor/extract.ts

```diff
@@ -281,18 +281,31 @@ export function fromProject(options: Options): Doc.DocPackage {
 
   return manager.data
 }
-
+let done: string[] = []
 /**
  * Recursively extract docs starting from exports of the given module.
  * Everything that is reachable will be considered.
  */
-export function fromModule(manager: Doc.Manager, sourceFile: tsm.SourceFile): Doc.DocPackage {
+export function fromModule(manager: Doc.Manager, sourceFile: tsm.SourceFile): Doc.Manager {
   const mod = Doc.modFromSourceFile(manager, sourceFile)
-
-  for (const ex of sourceFile.getExportedDeclarations()) {
-    const exportName = ex[0]
-    const n = ex[1][0]
-    const t = n.getType()
+  const exportedDeclarations = sourceFile.getExportedDeclarations()
+  debugExport(`filepath: ${sourceFile.getFilePath()}`)
+  debugExport({ exportedDeclarations, mod })
+  for (const [exportName, expDeclarations] of exportedDeclarations) {
+    let n = expDeclarations[0]
+    debugExport('-> export declaration %s', exportName)
+    debugExport(n)
+
+    debugExport('-> export declaration %s', n.getKindName())
+    if (tsm.Node.isSourceFile(n)) {
+      if (!done.includes(n.getFilePath())) {
+        done.push(n.getFilePath())
+        manager = fromModule(manager, n)
+      }
+      continue
+    }
```

Comment:
What was this about?

Reply:
If the node is a SourceFile run `fromModule` then continue to the next iteration in the for loop

Reply:
> If the node is a SourceFile run fromModule

I guess I'm wondering how that could be since we're dealing with exports. Like I'm guessing there is a pattern I haven't thought of/realized... Curious to learn more about this case in practice. I guess fp-ts has one!?

Reply:
I think it's when you have something like `import * as this from 'that'`

Reply:
A `SourceFile` Node type is different than imports as far as I know.